### PR TITLE
Collections index app

### PR DESCRIPTION
**Pull request by eessex**
Sets up `adminOnly` route for `/collections` index page.

TODO: test the index route.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -53,7 +53,7 @@
     "@artsy/express-reloadable": "^1.3.1",
     "@artsy/palette": "^2.18.1",
     "@artsy/passport": "^1.1.0",
-    "@artsy/reaction": "^5.11.0",
+    "@artsy/reaction": "^6.0.0",
```
Comment (yuki24) [inline package.json]: I wonder if this version upgrade should separately be deployed to be safe.

cc @alloy

Comment (eessex) [reply package.json]: Yeah sounds like it is happening separately, this is on hold to wait until the upgrade is merged.

**Comment by eessex** [comment]
@yuki24 @l2succes ok to merge?

File: src/desktop/apps/collections/client.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { buildClientApp } from "reaction/Artsy/Router/client"
+import { data as sd } from "sharify"
+import { routes } from "reaction/Apps/Collections/routes"
+import mediator from "desktop/lib/mediator.coffee"
+import React from "react"
+import ReactDOM from "react-dom"
+
+buildClientApp({ routes, user: sd.CURRENT_USER })
```
Comment (damassi) [inline src/desktop/apps/collections/client.js]: `user` should be passed into `context`: 
```
buildClientApp({ routes, context: { user: sd.CURRENT_USER }})
```

File: src/desktop/apps/collections/client.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  .then(({ ClientApp }) => {
+    ReactDOM.hydrate(
+      <ClientApp mediator={mediator} />,
```
Comment (damassi) [inline src/desktop/apps/collections/client.js]: This pattern changed a few months back -- `mediator` is also passed into `context`.

File: src/desktop/apps/collections/server.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    // Render layout
+    const layout = await stitch({
+      basePath: __dirname,
+      layout: "../../components/main_layout/templates/react_redesign.jade",
+      config: {
+        styledComponents: true,
+      },
+      blocks: {
+        head: () => null,
+        body: () => <ServerApp />,
```
Comment (damassi) [inline src/desktop/apps/collections/server.js]: `body: ServerApp`

File: src/desktop/apps/collections/client.js (hunk 3 above)
Comment (damassi) [reply src/desktop/apps/collections/client.js]: (Anything you want accessible in `SystemContext` is passed into `context`)

File: src/desktop/apps/collections/server.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+const app = (module.exports = express())
+
+app.get("/collections", adminOnly, async (req, res, next) => {
+  try {
+    const { IS_MOBILE } = res.locals.sd
+
+    const { ServerApp, redirect } = await buildServerApp({
+      routes,
+      url: req.url,
+      context: buildServerAppContext(req, res),
```
Comment (l2succes) [inline src/desktop/apps/collections/server.js]: didn't know about `buildServerAppContext`, nice

**Comment by eessex** [comment]
@l2succes the meta is included in the reaction component, what further steps are necessary for the Head?

**Pull request merged**